### PR TITLE
bump viem

### PR DESCRIPTION
**Pull request by andrewkmin**
## Summary & Motivation
bump our upstream dependency, https://github.com/wevm/viem.

note that there aren't many that affect our turnkey viem package directly; most pertain to OP chains, cryptography, etc. however, there are some minor interface changes: see changes to examples/tests.

additionally, you can find all releases here: https://github.com/wevm/viem/releases

## How I Tested These Changes
- unit
- examples

## Did you add a changeset?
yeeeeees

If updating one of our packages, you'll likely need to add a changeset to your PR. To do so, run [`pnpm changeset`](https://pnpm.io/using-changesets#adding-new-changesets). `pnpm changeset` will generate a file where you should write a human friendly message about the changes. Note how this ([example](https://github.com/tkhq/sdk/blob/b409cd06790f011bf939adcf0755499b8e7497ae/.changeset/extra-http-exports.md?plain=1#L1)) includes the package name (should be auto added by the command) along with the type of [semver change (major.minor.patch)](https://semver.org/) (which you should set).

These changes will be used at release time to determine what packages to publish and how to bump their version. For more context see [this comment](https://github.com/tkhq/sdk/pull/67#issuecomment-1568838440).

### CODE REVIEW

File: packages/viem/package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "compile:contracts": "hardhat compile"
   },
   "peerDependencies": {
-    "viem": "^1.16.6"
+    "viem": "^2.1.1"
   },
```
Comment (r-n-o) [inline packages/viem/package.json:56]: Is there a way to be lax here? (basically declaring: "turnkey/viem is compatible with 1.x or 2.x")

I'm afraid that consumers of turnkey/viem who do not want to be on 2.x won't be able to use the new turnkey/viem.

Comment (r-n-o) [reply packages/viem/package.json:56]: Something like `"viem": "^1.16.6 || ^2.1.1"`?

Comment (andrewkmin) [reply packages/viem/package.json]: good call! https://github.com/tkhq/sdk/pull/197/commits/850634fb27b2c2dfc0b6e1c63bc5d5fe9c7971ba

i bumped the with-viem example's viem dep back down to 1.16.6, and all seems to be well.

Comment (r-n-o) [reply packages/viem/package.json]: You can bump the with-viem example to 2.x I think!

Either way we won't have everything covered: if you leave it at 1.x we ensure compatibility with Viem 1.x, if you bump to 2.x we test for viem 2.x -- either way we lose one side :)

**Review by r-n-o — APPROVED** [review]

**Pull request merged**